rcl/src/rcl/arguments.c

```diff
@@ -354,6 +354,45 @@ rcl_get_zero_initialized_arguments(void)
   return default_arguments;
 }
 
+rcl_ret_t
+rcl_remove_ros_arguments(
+  char const * const argv[],
+  rcl_arguments_t * args,
+  rcl_allocator_t allocator,
+  int * nonros_argc,
+  const char ** nonros_argv[])
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(argv, RCL_RET_INVALID_ARGUMENT, allocator);
+  RCL_CHECK_ARGUMENT_FOR_NULL(nonros_argc, RCL_RET_INVALID_ARGUMENT, allocator);
+  RCL_CHECK_ARGUMENT_FOR_NULL(args, RCL_RET_INVALID_ARGUMENT, allocator);
+  RCL_CHECK_ALLOCATOR_WITH_MSG(&allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
```

Comment:
You should probably check the allocator first, otherwise you're passing a potentially invalid allocator to the other null check macros.